### ISSUE TITLE
[FLANG][OpenMP] flang fails to detect error of zero-sized array in DEPEND clause.

### ISSUE THREAD

**Issue by laoshd** [llvm/llvm-project#222551]
OpenMp 5.0 section 2.17.11 depend Clause: Restrictions say "List items used in depend clauses cannot be zero-length array sections." A compiler compiling the following code should produce an error message. Flang however accepts the code silently:

`% cat tst11.f90`

```fortran
INTEGER, DIMENSION(0) :: x
!$OMP TASKWAIT DEPEND(IN:x(:))
END
```

```console
% flang -fopenmp -fopenmp-version=50  tst11.f90
warning: OpenMP support for version 50 in flang is still incomplete [-Wexperimental-option]
```

**Comment by laoshd** [comment]
I'm interested in this problem. Would you please assign it to me @laoshd? Thanks.
